STAC-23602: Documentation for the new backup cli tool (#141)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/data-management/backup_restore/backup_enable.adoc

= Enable backups
:revdate: 2025-07-10
:page-revdate: {revdate}
:description: SUSE Observability Self-hosted

[NOTE]
====

> This page applies to SUSE Observability vX.Y.Z or newer.

Reviewer: ```suggestion
This page applies to {stackstate-product-name} vX.Y.Z or newer.
```

====

== Overview

SUSE Observability has a built-in backup mechanism that can be configured to store backups to AWS S3, Azure Blob Storage, or a Kubernetes Persistent Volume.

> Most backups are enabled with a single Helm value `global.backup.enabled`. Settings backups are always enabled but behave differently based on this value (see below).

Reviewer: By "this value", do we mean the below sub-sections, such as "Backup scope" and "storage options"?

If yes, then I think it's better to end the statement with a colon.

```suggestion
Most backups are enabled with a single Helm value `global.backup.enabled`. Settings backups are always enabled but behave differently based on the following values:
```

=== Backup scope

The following data can be automatically backed up:

* *Configuration and topology data* stored in StackGraph - enabled when `global.backup.enabled` is `true`
* *Settings* (StackPacks, monitors, views, tokens) - always enabled:
** When `global.backup.enabled` is `true`: backups are stored via MinIO to your configured storage backend
** When `global.backup.enabled` is `false`: backups are stored to a dedicated Kubernetes Persistent Volume, bypassing MinIO
* *Metrics* stored in SUSE Observability's Victoria Metrics instance(s) - enabled when `global.backup.enabled` is `true`
* *Telemetry data* stored in SUSE Observability's Elasticsearch instance - enabled when `global.backup.enabled` is `true`

> * *OpenTelemetry data* stored in SUSE Observability's ClickHouse instance - enabled when `global.backup.enabled` is `true`

Reviewer: ```suggestion
* *OpenTelemetry data* stored in {stackstate-product-name}'s ClickHouse instance - enabled when `global.backup.enabled` is `true`
```

  elasticsearch:
    bucketName: AWS_ELASTICSEARCH_BUCKET
  configuration:
    bucketName: AWS_CONFIGURATION_BUCKET
victoria-metrics-0:
  backup:
    bucketName: AWS_VICTORIA_METRICS_BUCKET
victoria-metrics-1:
  backup:
    bucketName: AWS_VICTORIA_METRICS_BUCKET
clickhouse:
  backup:
    bucketName: AWS_CLICKHOUSE_BUCKET
minio:
  accessKey: YOUR_ACCESS_KEY
  secretKey: YOUR_SECRET_KEY
  s3gateway:
    enabled: true
    accessKey: AWS_ACCESS_KEY
    secretKey: AWS_SECRET_KEY
----

Replace the following values:

* `YOUR_ACCESS_KEY` and `YOUR_SECRET_KEY` are the credentials that will be used to secure the MinIO system. These credentials are set on the MinIO system and used by the automatic backup jobs and the restore jobs. They're also required if you want to manually access the MinIO system.
 ** YOUR_ACCESS_KEY should contain 5 to 20 alphanumerical characters.
 ** YOUR_SECRET_KEY should contain 8 to 40 alphanumerical characters.
* `AWS_ACCESS_KEY` and `AWS_SECRET_KEY` are the AWS credentials for the IAM user that has access to the S3 buckets where the backups will be stored. See below for the permission policy that needs to be attached to that user.

> * `AWS_*_BUCKET` are the names of the S3 buckets where the backups should be stored. Note: The names of AWS S3 buckets are global across the whole of AWS, therefore the S3 buckets with the default name (`sts-elasticsearch-backup`, `sts-configuration-backup`, `sts-stackgraph-backup`, `sts-victoria-metrics-backup` and `sts-clickhouse-backup` ) will probably not be available.

Reviewer: ```suggestion
* `AWS_*_BUCKET` are the names of the S3 buckets where the backups should be stored. 
+
[Note]
====
The names of AWS S3 buckets are global across the whole of AWS. Therefore, the S3 buckets, with the default name (`sts-elasticsearch-backup`, `sts-configuration-backup`, `sts-stackgraph-backup`, `sts-victoria-metrics-backup` and `sts-clickhouse-backup` ), will probably not be available.
====
```

==== Using a single S3 bucket with prefixes

Instead of using separate buckets for each datastore, you can use a single S3 bucket with different prefixes:

[,yaml]
----
global:
  backup:
    enabled: true
backup:
  elasticsearch:
    bucketName: BUCKET
    s3Prefix: elasticsearch
  stackGraph:
    bucketName: BUCKET
    s3Prefix: stackgraph
  configuration:
    bucketName: BUCKET
    s3Prefix: configuration
victoria-metrics-0:
  backup:
    bucketName: BUCKET
    s3Prefix: victoria-metrics-0
victoria-metrics-1:
  backup:
    bucketName: BUCKET
    s3Prefix: victoria-metrics-1
clickhouse:
  backup:
    bucketName: BUCKET
    s3Prefix: clickhouse
minio:
  accessKey: YOUR_ACCESS_KEY
  secretKey: YOUR_SECRET_KEY
  s3gateway:
    enabled: true
    accessKey: AWS_ACCESS_KEY
    secretKey: AWS_SECRET_KEY
----

> Replace `BUCKET` with your S3 bucket name. The backups for different datastores will be organized using the configured `s3Prefix` values. The same `YOUR_ACCESS_KEY`, `YOUR_SECRET_KEY`, `AWS_ACCESS_KEY`, and `AWS_SECRET_KEY` values from the previous section apply here.

Reviewer: ```suggestion
Replace `BUCKET` with your S3 bucket name. The backups for different datastores are organized using the configured `s3Prefix` values. The same `YOUR_ACCESS_KEY`, `YOUR_SECRET_KEY`, `AWS_ACCESS_KEY`, and `AWS_SECRET_KEY` values from the previous section apply here.
```

    bucketName: CONTAINER
    s3Prefix: configuration
victoria-metrics-0:
  backup:
    bucketName: CONTAINER
    s3Prefix: victoria-metrics-0
victoria-metrics-1:
  backup:
    bucketName: CONTAINER
    s3Prefix: victoria-metrics-1
clickhouse:
  backup:
    bucketName: CONTAINER
    s3Prefix: clickhouse
minio:
  accessKey: AZURE_STORAGE_ACCOUNT_NAME
  secretKey: AZURE_STORAGE_ACCOUNT_KEY
  azuregateway:
    enabled: true
----

Replace `CONTAINER` with your Azure Blob Storage container name. The backups for different datastores will be organized using the configured `s3Prefix` values. The same `AZURE_STORAGE_ACCOUNT_NAME` and `AZURE_STORAGE_ACCOUNT_KEY` values from the previous section apply here.

=== Kubernetes Persistent Volume

[CAUTION]
====
Using Kubernetes Persistent Volumes for backups has significant limitations:

* Expensive - Cloud providers typically use block storage (EBS/Azure Block) which is costly for large backups
* No disaster recovery - PVs are destroyed if the cluster is deleted
* Not portable - Cannot restore backups to a different cluster

**Recommendation:** Use AWS S3 or Azure Blob Storage instead for production environments.
====

==== Basic configuration

> To enable backups to cluster-local storage, enable MinIO by adding the following YAML fragment to the Helm `values.yaml` file used to install SUSE Observability:

Reviewer: ```suggestion
To enable backups to cluster-local storage, enable MinIO by adding the following YAML fragment to the Helm `values.yaml` file used to install {stackstate-product-name}:
```

    enabled: true
----

Replace the following values:

* `YOUR_ACCESS_KEY` and `YOUR_SECRET_KEY` - the credentials that will be used to secure the MinIO system. The automatic backup jobs and the restore jobs will use them. They're also required to manually access the MinIO storage. `YOUR_ACCESS_KEY` should contain 5 to 20 alphanumerical characters and `YOUR_SECRET_KEY` should contain 8 to 40 alphanumerical characters.

== Configuration and topology data (StackGraph)

Configuration and topology data (StackGraph) backups are full backups, stored in a single file with the extension `.graph`. Each file contains a full backup and can be moved, copied or deleted as required.

=== Backup schedule

By default, the StackGraph backups are created daily at 03:00 AM server time.

The backup schedule can be configured using the Helm value `backup.stackGraph.scheduled.schedule`, specified in https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#_cron_schedule_syntax[Kubernetes cron schedule syntax (kubernetes.io)].

=== Backup retention

By default, the StackGraph backups are kept for 30 days. As StackGraph backups are full backups, this can require a lot of storage.

> The backup retention delta can be configured using the Helm value `backup.stackGraph.scheduled.backupRetentionTimeDelta`, specified in the format of GNU date `--date` argument. For example, the default is `30 days ago`. See https://www.gnu.org/software/coreutils/manual/html_node/Relative-items-in-date-strings.html[Relative items in date strings] for more examples.

Author: ```suggestion
The backup retention delta can be configured using the Helm value `backup.stackGraph.scheduled.backupRetentionTimeDelta`, specified in the format of GNU date `--date` argument. The default is `30 days ago`. See https://www.gnu.org/software/coreutils/manual/html_node/Relative-items-in-date-strings.html[Relative items in date strings] for more examples.
```
